### PR TITLE
Update `gpus` flag with `accelerator` and `devices` flag

### LINKED ISSUE #11833: update all docs to reflect new syntax for accelerators
All the docs still say:
```python
Trainer(gpus=...)
```

But we now support a different syntax:
```python
Trainer(devices=4, accelerator='gpu')
```

Although we have backward support, all the docs (including readme) needs to reflect the new way of doing it

cc @borda @rohitgr7 @akihironitta @ananthsub @carmocca @Borda

### PR DESCRIPTION
**Pull request by kaushikb11**
## What does this PR do?

Update `gpus` flag with `accelerator` and `devices` flag
Fixes #11833 

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda @rohitgr7 @justusschock @awaelchli @kaushikb11 @akihironitta

### CODE REVIEW

**Review by ananthsub — APPROVED** [review]

File: docs/source/accelerators/gpu.rst [hunk 1]
```diff
@@ -718,15 +718,15 @@ Below are the possible configurations we support.
 +-------+---------+----+-----+--------+------------------------------------------------------------+
```
Comment (rohitgr7) [inline docs/source/accelerators/gpu.rst:718]: I think this table needs a little more formatting

File: docs/source/accelerators/gpu.rst [hunk 2]
```diff
@@ -207,23 +207,23 @@ a comma separated list of GPU ids:
...
     # To use all available GPUs put -1 or '-1'
     # equivalent to list(range(torch.cuda.device_count()))
-    Trainer(gpus=-1)
+    Trainer(accelerator="gpu", devices=-1)
 
 The table below lists examples of possible input formats and how they are interpreted by Lightning.
-Note in particular the difference between `gpus=0`, `gpus=[0]` and `gpus="0"`.
+Note in particular the difference between `devices=0`, `devices=[0]` and `devices="0"`.
 
 +---------------+-----------+---------------------+---------------------------------+
```
Comment (rohitgr7) [inline docs/source/accelerators/gpu.rst:228]: update this table too?

**Review by rohitgr7 — APPROVED** [review]

File: docs/source/accelerators/gpu.rst (hunk 2 above)
Comment (carmocca) [reply docs/source/accelerators/gpu.rst:228]: +1

**Review by carmocca — APPROVED** [review]

File: docs/source/accelerators/gpu.rst [hunk 3]
```diff
@@ -207,23 +207,23 @@ a comma separated list of GPU ids:
...
 | `gpus`        | Type      | Parsed              | Meaning                         |
```
Comment (kaushikb11) [inline docs/source/accelerators/gpu.rst:229]:
```suggestion
| `devices`        | Type      | Parsed              | Meaning                         |
```

Comment (rohitgr7) [reply docs/source/accelerators/gpu.rst:229]: don't go that easy.. fix the formatting xD

Comment (kaushikb11) [reply docs/source/accelerators/gpu.rst]: Hah, figured. done

File: docs/source/accelerators/gpu.rst (hunk 2 above)
Comment (kaushikb11) [reply docs/source/accelerators/gpu.rst]: Done!

File: docs/source/accelerators/gpu.rst [hunk 4]
```diff
@@ -207,51 +207,51 @@ a comma separated list of GPU ids:
...
-+---------------+-----------+---------------------+---------------------------------+
-| "0"           | str       | None                | CPU                             |
-+---------------+-----------+---------------------+---------------------------------+
-| "3"           | str       | [0, 1, 2]           | first 3 GPUs                    |
-+---------------+-----------+---------------------+---------------------------------+
-| "1, 3"        | str       | [1, 3]              | GPUs 1 and 3                    |
-+---------------+-----------+---------------------+---------------------------------+
-| "-1"          | str       | [0, 1, 2, ...]      | all available GPUs              |
-+---------------+-----------+---------------------+---------------------------------+
+Note in particular the difference between `devices=0`, `devices=[0]` and `devices="0"`.
```
Comment (rohitgr7) [inline docs/source/accelerators/gpu.rst:226]:
```suggestion
Note in particular the difference between ``devices=0``, ``devices=[0]`` and ``devices="0"``.
```

File: docs/source/common/trainer.rst [hunk 5]
```diff
@@ -695,6 +696,9 @@ See Also:
 gpus
 ^^^^
 
+.. warning:: Setting `Trainer(gpus=x)` is deprecated in v1.6 and will be removed"
+    in v2.0. Please use `Trainer(accelerator='gpu', devices=x)` instead.
+
```
Comment (awaelchli) [inline docs/source/common/trainer.rst:701]: this is not yet the case. let's not add it until we have the deprecation warnings

**Review by awaelchli — APPROVED** [review]

**Review by Borda — APPROVED** [review]

File: tests/strategies/test_ddp.py [hunk 6]
```diff
@@ -29,7 +29,7 @@
 from tests.strategies import ddp_model
 from tests.utilities.distributed import call_training_script
 
-CLI_ARGS = "--max_epochs 1 --gpus 2 --strategy ddp"
+CLI_ARGS = "--max_epochs 1 --accelerator 'gpu' --devices 2 --strategy ddp"
```
Comment (kaushikb11) [inline tests/strategies/test_ddp.py:32]:
```suggestion
CLI_ARGS = "--max_epochs 1 --accelerator gpu --devices 2 --strategy ddp"
```

**Review by ananthsub — APPROVED** [review]
